### PR TITLE
refactor/DYN-6124/code-smells

### PR DESCRIPTION
**Pull request by Enzo707**
after-code-smells-fixes-min (3)

This PR includes refactoring due to code smells reports.

- Removes useless functions and commented legacy code
- Updates according to ES6+ and javaScript recommended standards

**FYI**
@QilongTang 
@RobertGlobant20

### CODE REVIEW

File: src/LibraryUtilities.ts [hunk 1]
```diff
@@ -492,13 +489,10 @@ export function buildLibrarySectionsFromLayoutSpecs(loadedTypes: any, layoutSpec
         sectionElements.push(new LayoutElement(section));
     }
 
-    //call update category groups.
-    updateCategoryGroups(sectionElements);
```
Comment (RobertGlobant20) [inline src/LibraryUtilities.ts:496]: is this call not necesary? or is done in other place?

File: src/LibraryUtilities.ts [hunk 2]
```diff
@@ -825,7 +784,7 @@ export function findAndExpandItemByPath(pathToItem: ItemData[], allItems: ItemDa
     );
 
     if (pathToItem.length == 1) {
-        return item ? true : false;
+        return !!item;
```
Comment (RobertGlobant20) [inline src/LibraryUtilities.ts:787]: could you explain why` item ? true : false; `is the same than` !!item;`?
I know that "!" is the logical not operator but not sure why is used two times?

**Review by RobertGlobant20 — COMMENTED** [review]
LGTM with two comments.
Also did you test this changes in DynamoSandbox? Please include a GIF of your testing.

File: src/LibraryUtilities.ts (hunk 1 above)
Comment (Enzo707) [reply src/LibraryUtilities.ts:496]: @RobertGlobant20 Please, let me know if I'm missing something but If you look at the function declaration, it doesn't return anything that might affect the context where it is executed. Also, all the operations there are related to `sectionElements` which is defined as a `let`.

File: src/LibraryUtilities.ts (hunk 2 above)
Comment (Enzo707) [reply src/LibraryUtilities.ts:787]: @RobertGlobant20 !! operator returns the boolean value. [Ref](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_NOT#double_not_!!)

File: src/LibraryUtilities.ts [hunk 3]
```diff
@@ -116,7 +114,6 @@ export class ItemData {
         }
 
         // Get keywords from typeListNode and push them into itemData
-        let keywords = typeListNode.keywords.split(",");
```
Comment (QilongTang) [inline src/LibraryUtilities.ts:119]: why do we remove this line?

Comment (Enzo707) [reply src/LibraryUtilities.ts:119]: @QilongTang This variable is not used. The operations below this statement refers to keywords property of class ItemData class (`this.keywords`)

File: src/components/LibraryItem.tsx [hunk 4]
```diff
@@ -45,31 +44,15 @@ class GroupedItems {
...
-                case "query": this.queries.push(items[i]); break;
-                default: this.others.push(items[i]); break;
+            switch (element.itemType) {
+                case "create": this.creates.push(element); break;
+                case "action": this.actions.push(element); break;
+                case "query": this.queries.push(element); break;
+                default: this.others.push(element); break;
             }
 
-            var finalData = {
```
Comment (QilongTang) [inline src/components/LibraryItem.tsx:57]: I guess this is also not used?

File: src/LibraryUtilities.ts (hunk 1 above)
Comment (QilongTang) [reply src/LibraryUtilities.ts:496]: yeah looks like the function was disabled so no longer valid

**Review by QilongTang — COMMENTED** [review]
LGTM with one comment

File: src/components/LibraryItem.tsx (hunk 4 above)
Comment (Enzo707) [reply src/components/LibraryItem.tsx:57]: Right, it is not used.

**Review by QilongTang — APPROVED** [review]

**Comment by QilongTang** [comment]
Please make the change to update librarie.js on Dynamo `master` branch only

**Pull request merged**